src/main/java/com/conveyal/r5/analyst/PointSetCache.java

```diff
@@ -0,0 +1,75 @@
+package com.conveyal.r5.analyst;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.Map;
+
+/**
+ * A cache for a geographic PointSet, including grid-based and custom pointset-based sets.
+ * This allows pointsets to not be recreated and relinked on every analysis task.
+ * The cache does not have expiration.
+ */
+public class PointSetCache {
```

Comment:
See comment on PointSetWithIds below. In our eventual design we intend to have a single abstract PointSet class or interface and the fact that it's gridded or free-form would be an implementation detail, with gridded points having implicit coordinates.